cloud/amazon/ec2_vpc_managed_ngw_facts.py

```diff
@@ -0,0 +1,162 @@
+        for possible filters.
+  required: false
+  default: None
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Simple example of listing all nat gateways
+- name: List all managed nat gateways in ap-southeast-2
+  ec2_vpc_managed_ngw_facts:
+    region: ap-southeast-2
+  register: all_ngws
+
+- name: Debugging the result
+  debug:
+    msg: "{{ all_ngws.result }}"
+
+- name: Get details on specific nat gateways
+  ec2_vpc_managed_ngw_facts:
+    nat_gateway_ids:
+      - nat-1234567891234567
+      - nat-7654321987654321
+    region: ap-southeast-2
+  register: specific_ngws
+
+- name: Get all nat gateways with specific filters
+  ec2_vpc_managed_ngw_facts:
+    region: ap-southeast-2
+    filters:
+      state: ['pending']
+  register: pending_ngws
+
+- name: Get nat gateways with specific filter
+  ec2_vpc_managed_ngw_facts:
+    region: ap-southeast-2
+    describe_filter:
+      subnet-id: subnet-12345678
+      state: ['available']
+  register: existing_nat_gateways
+'''
+
+RETURN = '''
+result:
+  description: The result of the describe.
+  returned: success
+  type: list
+'''
+
+try:
+    import json
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import time
+
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+
+def get_nat_gateways(client, module, nat_gateway_id=None):
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filter'] = []
+        for key, value in module.params.get('filters').iteritems():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filters'].append(temp_dict)
+    if module.params.get('nat_gateway_ids'):
+        params['NatGatewayIds'] = module.params.get('nat_gateway_ids')
+
+    try:
+        result = json.loads(json.dumps(client.describe_nat_gateways(**params), default=date_handler))
+    except Exception as e:
+        module.fail_json(msg=str(e.message))
+
+    return result['NatGateways']
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        filters=dict(default=None, type='dict'),
+        nat_gateway_ids=dict(default=None, type='list'),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    # Validate Requirements
+    if not HAS_BOTO3:
+        module.fail_json(msg='json and botocore/boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        ec2 = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except Exception as e:
+        # Getting around the get_aws_connection_info boto reliance for region
```

Comment:
Hi @willthames. 
Yes - it is a little ugly unfortunately. Basically as we are trying to use boto3 and botocore - from discussions on other modules previously. 
The problem with this and not importing boto is the module_utils/ec2 with the function get_aws_connection_info() which for the region check falls through to the else statement for users that want to have their default regions set inside the standard $HOME/.aws/config. 
This would normally be fine but if you look at that else statement, it's using boto and as we aren't importing boto it errors. There was a PR raised  quite a while ago which is still waiting to be merged. As a result, I'm popping this in as a workaround. I can update it to catch NameError (and then check if the message exception matches) and then have another except for the normal botocore.exceptions.NoCredentialsError as well.